packages/nodes-base/nodes/SplitInBatches/v3/SplitInBatchesV3.node.ts

```diff
@@ -5,9 +5,39 @@ import type {
 	INodeTypeDescription,
 	IPairedItemData,
 } from 'n8n-workflow';
-import { NodeConnectionTypes, deepCopy } from 'n8n-workflow';
+import { NodeConnectionTypes, deepCopy, NodeOperationError } from 'n8n-workflow';
 
 export class SplitInBatchesV3 implements INodeType {
+	// Global tracking across all instances within the same workflow execution
+	private static executionCounters = new Map<string, number>();
+
+	// Track executions for automatic cleanup after timeout
+	private static executionTracking = new Map<string, number>();
+
+	// Timer reference for cleanup
+	private static cleanupTimer: NodeJS.Timeout | null = null;
+
+	static {
+		// Start periodic cleanup for abandoned executions (every 5 minutes)
+		SplitInBatchesV3.cleanupTimer = setInterval(
+			() => {
+				const now = Date.now();
+				const timeout = 5 * 60 * 1000; // 5 minutes
+
+				for (const [executionId, timestamp] of SplitInBatchesV3.executionTracking) {
+					if (now - timestamp > timeout) {
```

Comment:
Not sure if this will be reliable, workflows could take longer than 5mins between loops, especially with Wait nodes or HITL.